components/asp/database/SystemWriter.py

```
@@ -79,10 +84,17 @@ def addToSystemIndex(self, header):
             Adds sys info to SYSTEMTABLES and creates tables for programs, deployments and traces.
         '''
         systemInfo = header.sysinfo
-        system_id = systemInfo["metadata"]["systemId"]        
-        system_ver = systemInfo["metadata"]["systemVersion"]
-        name = systemInfo["metadata"]["name"]
-        description = systemInfo["metadata"]["description"]
+        if systemInfo and "metadata" in systemInfo and header.sysinfo["metadata"]:
+            system_id = header.sysinfo["metadata"]["systemId"] if "systemId" in header.sysinfo["metadata"] else None
```

Ruff (0.11.9)

Use `header.sysinfo["metadata"].get("systemId", None)` instead of an `if` block

Replace with `header.sysinfo["metadata"].get("systemId", None)`

(SIM401)